src/Aurora.Infrastructure/Scrapers/XvideosScraper.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Aurora.Application.Models;
+using Aurora.Infrastructure.Contracts;
+using Aurora.Infrastructure.Services;
+using Aurora.Shared.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Aurora.Infrastructure.Scrapers
+{
+    public class XvideosScraper : ScraperBase
+    {
+        private const int PAGE_NUMBER_LIMIT = 5;
+        private const string _baseUrl = "https://www.xvideos.com";
+        private readonly IWebClientService _clientProvider;
+        private readonly DriverInitializer _initializer;
+
+        public XvideosScraper(ILogger<XvideosScraper> logger, IWebClientService clientProvider,
+            DriverInitializer initializer) : base(logger)
+        {
+            _clientProvider = clientProvider;
+            _initializer = initializer;
+        }
+
+        public override async Task<ValueOrNull<SearchResult>> SearchInner(
+            SearchRequest request, CancellationToken token = default)
+        {
+            SearchResult result = new()
+            {
+                Website = _baseUrl
+            };
+
+            try
+            {
+                if (request.SearchOptions.Contains(SearchOption.Video))
+                {
+                    var videos = await ScrapVideos(request.SearchTerm, request.ResponseItemsMaxCount);
+                    result.Items.AddRange(videos);
+                }
+
+                if (request.SearchOptions.Contains(SearchOption.Gif))
+                {
+                    var gifs = await ScrapGifs(request.SearchTerm, request.ResponseItemsMaxCount);
+                    result.Items.AddRange(gifs);
+                }
+
+                if (request.SearchOptions.Contains(SearchOption.Image))
+                {
+                    var images = await ScrapImages(request.SearchTerm, request.ResponseItemsMaxCount);
+                    result.Items.AddRange(images);
+                }
+            }
+            catch (Exception exception)
+            {
+                return ValueOrNull<SearchResult>.CreateNull(exception.Message);
+            }
+
+            result.CountItems = result.Items.Count;
+            return ValueOrNull<SearchResult>.CreateValue(result);
```

Comment:
You can just return result, which would be implicitly converter to valur or null